### PR TITLE
Add option to change "=" symbol in ModelCheckpoint filenames

### LINKED ISSUE #17825: Option to remove "=" sign from `ModelCheckpoint` ckpt file names
### Description & Motivation

Currently `ModelCheckpoint` saves checkpoints with filenames in the form: `epoch=01-metric=0.4367.ckpt`

The "=" sign can create issues for services where the checkpoint is uploaded, e.g. Azure/MLFlow or Kaggle Datasets since I think there are more restrictions on allowed characters. I think the issue is that if the file name ever ends up in a URL, the "=" could be confused with a query param.

Currently, I have a helper script to rename my files before upload, but it would be good if `ModelCheckpoint` could have the option to prevent this

### Pitch

Some possible solutions:

- Use an underscore instead of "="?
- Don't use a character at all, e.g. `epoch01-metric0.4367.ckpt`
- An option for the user to define the character in the kwargs?

### Alternatives

_No response_

### Additional context

_No response_

cc @borda @awaelchli @carmocca

### PR DESCRIPTION
**Pull request by giorgioskij**
## What does this PR do?

Part of #17825

This PR adds a feature, as discussed in issue #17825, to change the default "=" symbol between the name of a field and its value in the filenames generated by the ModelCheckpoint class. There was already an option to change the separator symbol in the filenames, so this follows the same exact implementation style.

Example:
```
ModelCheckpoint.CHECKPOINT_JOIN_CHAR = "-"  # this was already present
ModelCheckpoint.CHECKPOINT_EQUALS_CHAR = ":"  # this is my addition
ckp = ModelCheckpoint(dirpath=".", monitor="valid_loss")
trainer = L.Trainer(callbacks=[ckp])
```
now this trainer will save its checkpoints as: `epoch:1-step:1234.ckpt` instead of `epoch=1-step=1234.ckpt`

While adding the new attribute to the docstring of the class ModelCheckpoint, we noticed a discrepancy between code and docstring. The docs say:
```
For example, you can change the default last checkpoint name by doing
``checkpoint_callback.CHECKPOINT_NAME_LAST = "{epoch}-last"``
```
However, the correct way to change that attributes at the moment is by using the class name, and not the instance name, so:
`ModelCheckpoint.CHECKPOINT_NAME_LAST = {epoch}-last`

In issue #17825 we decided to keep this as is, and make a new pr to change the code to behave how it is currently documented.

I also updated CHANGELOG.md to mention this new attribute, including the link to this PR.

Thanks a lot for your time reviewing this!

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+- Added `CHECKPOINT_EQUALS_SYMBOL` attribute to `ModelCheckpoint` class ([#17999](https://github.com/Lightning-AI/lightning/pull/17999))
```
Comment (awaelchli) [inline src/lightning/pytorch/CHANGELOG.md:12]:
```suggestion
- Added `CHECKPOINT_EQUALS_CHAR` attribute to `ModelCheckpoint` class ([#17999](https://github.com/Lightning-AI/lightning/pull/17999))
```

**Review by awaelchli — COMMENTED** [review]
Nice addition!
We would need a test too. I suggest to extend the existing `test_model_checkpoint_format_checkpoint_name` in test_model_checkpoint.py.

File: src/lightning/pytorch/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -523,7 +525,7 @@ def _format_checkpoint_name(
             name = group[1:]
 
             if auto_insert_metric_name:
-                filename = filename.replace(group, name + "={" + name)
+                filename = filename.replace(group, name + cls.CHECKPOINT_EQUALS_CHAR + "{" + name)
```
Comment (awaelchli) [inline src/lightning/pytorch/callbacks/model_checkpoint.py:528]: @carmocca's suggestion was to also replace cls here with self (removing the `@classmethod` decorator and making it a regular method. Could be done here or in a separate PR if you like.

Comment (giorgioskij) [reply src/lightning/pytorch/callbacks/model_checkpoint.py:528]: My only doubt is that by doing so I'm not sure of what's the point of keeping those values as class attributes instead of normal instance attributes, so I had proposed a separate PR to talk about the possibility of changing that as well.
But if you already know how you would like it best I can just change it to that right now

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
LGTM!
